lib/plugins/aws/package/compile/events/schedule/index.js

```diff
@@ -2,8 +2,11 @@
 
 const _ = require('lodash');
 
-const rateSyntaxPattern = /^rate\((?:1 (?:minute|hour|day)|(?:1\d+|[2-9]\d*) (?:minute|hour|day)s)\)$/;
-const cronSyntaxPattern = /^cron\(\S+ \S+ \S+ \S+ \S+ \S+\)$/;
+const rateSyntax = '^rate\\((?:1 (?:minute|hour|day)|(?:1\\d+|[2-9]\\d*) (?:minute|hour|day)s)\\)$';
+const cronSyntax = '^cron\\(\\S+ \\S+ \\S+ \\S+ \\S+ \\S+\\)$';
+const scheduleSyntax = `${rateSyntax}|${cronSyntax}`;
+
+const jsonPattern = '^\\{.*\\}$';
```

Comment:
It just says JSON string.  IIRC it turns in into an input object but it's a while since I've used this.  This should probably be either stricter or looser, but see the discussion for the mutually exclusive properties below.

Reply:
If you're sure that AWS will crash on non object JSON input (e.g. array, or string), then I think it's fine to keep it. Otherwise I would validate JSON string simply as _string_ without constraints.

Reply:
I'm not sure enough.  I think replacing it with simple type and length validation and letting AWS validate the content is probably safest.